### PR TITLE
ci: change to use GitHub actions

### PR DESCRIPTION
**Pull request by FrankYang0529**
#### Which issue(s) this PR fixes:
https://github.com/longhorn/longhorn/issues/8039

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-FROM registry.suse.com/bci/bci-base:15.5 AS build
+# syntax=docker/dockerfile:1.7.1
```
Comment (derekbit) [inline package/Dockerfile:1]: Ｗhat's the purpose of the external Dockerfile frontend?

File: scripts/build [hunk 2]
```diff
@@ -19,4 +19,5 @@ fi
 cd $(dirname $0)/..
 
 mkdir -p bin
-CGO_ENABLED=0 go build -o bin/longhorn-share-manager -ldflags "$LINKFLAGS" $COVER $COVERPKG
+CGO_ENABLED=0 GOARCH=amd64 go build -o bin/longhorn-share-manager-amd64 -ldflags "$LINKFLAGS" $COVER $COVERPKG
+CGO_ENABLED=0 GOARCH=arm64 go build -o bin/longhorn-share-manager-arm64 -ldflags "$LINKFLAGS" $COVER $COVERPKG
```
Comment (derekbit) [inline scripts/build:23]: Refine like
```
archs=("amd64" "arm64")
for arch in "${archs[@]}"; do
    CGO_ENABLED=0 GOARCH="$arch" go build -o "bin/longhorn-share-manager-$arch" -ldflags "$LINKFLAGS" $COVER $COVERPKG
done
```

File: package/Dockerfile (hunk 1 above)
Comment (FrankYang0529) [reply package/Dockerfile:1]: This one: https://github.com/rancher/support-bundle-kit/pull/110#discussion_r1569937958

File: scripts/build (hunk 2 above)
Comment (FrankYang0529) [reply scripts/build]: Updated it. Thanks.

**Review by innobead — APPROVED** [review]

**Comment by FrankYang0529** [comment]
@mergify backport v1.6.x

**Comment by FrankYang0529** [comment]
@mergify backport v1.5.x

**Pull request merged**